MvvmCross/Platforms/Ios/Binding/Views/MvxCollectionViewSourceAnimated.cs

```diff
@@ -44,6 +44,12 @@ public MvxCollectionViewSourceAnimated(UICollectionView collectionView, NSString
 
         protected override void CollectionChangedOnCollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
         {
+            if (!NSThread.IsMain)
+            {
+                InvokeOnMainThread(() => CollectionChangedOnCollectionChanged(sender, args));
```

Comment:
Can we maybe change this to BeginInvokeOnMainThread instead?

Reply:
No objection but curious what the difference is?

Reply:
BeginInvokeOnMainThread, won't block, while InvokeOnMainThread will block until the action is done.

Reply:
Done. Good to know :)